backends/bmv2/common/parser.cpp

```diff
@@ -25,7 +25,43 @@ limitations under the License.
 
 namespace BMV2 {
 
-cstring ParserConverter::jsonAssignment(const IR::Type *type, bool inParser) {
+/*
+Determines whether an lvalue is also a field expr.
+
+In BMv2 JSON, field expr will be converted to type "field" instead of "expression".
+
+Production rule for lvalue in P4:
+
+lvalue
+    : prefixedNonTypeName                { $$ = new IR::PathExpression($1); }
+    | THIS                               { $$ = new IR::This(@1); }
+    | lvalue dot_name %prec DOT          { $$ = new IR::Member(@1 + @2, $1, *$2); }
+    | lvalue "[" expression "]"          { $$ = new IR::ArrayIndex(@1 + @4, $1, $3); }
+    | lvalue "[" expression ":" expression "]" { $$ = new IR::Slice(@1 + @6, $1, $3, $5); }
+    ;
+
+To determine a field expr, we assume it corresponds to the following production rule:
+
+field_expr
+    : prefixedNonTypeName
+    | field_expr dot_name
+    | field_expr "[" constant "]"
+*/
+bool ParserConverter::isFieldExpr(const IR::Expression *expr) {
+    if (expr->is<IR::PathExpression>()) return true;
+    if (expr->is<IR::Member>()) {
+        auto member = expr->to<IR::Member>();
```

Comment:
nit: I think you can keep the code less verbose by using `if (auto member = expr->to<IR::Member>())`. I think the rest of the p4c codebase uses that a lot.

Reply:
Done.